rainforest/rainforest.go

```diff
@@ -113,7 +123,30 @@ func (c *Client) NewRequest(method, urlStr string, body interface{}) (*http.Requ
 
 	// Set UserAgent header with appended library version, will look like:
 	// "rainforest-cli/2.1.0 [rainforest golang lib/2.0.0]"
-	composedUserAgent := c.UserAgent + " [rainforest golang lib/" + libVersion + "]"
+	userAgent := []string{"rainforest", "golang", "lib/" + libVersion}
+
+	if c.SendTelemetry {
+		found, ci_name := detectci.WhichCI()
+		if found {
+			userAgent = append(userAgent, "ci/"+ci_name)
+		}
+
+		var remote string
+		git, err := gitTrigger.NewGitTrigger()
+		if err == nil {
+			remote, err = git.GetRemote()
+			if err == nil {
+				u, err := giturls.Parse(remote)
+				if err == nil {
+					// Strip the user details, if any
+					u.User = nil
```

Comment:
It'd be nice if we had a test to explicitly make sure this is working properly (and avoid shipping a change which accidentally results in sending the user details later on)